enhancements/baremetal/baremetal.md

```diff
@@ -0,0 +1,283 @@
+---
+title: Adding Baremetal Installer Provisioned Infrastructure (IPI) to OpenShift
+authors:
+  - "@sadasu"
+reviewers:
+  - "@smarterclayton"
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+approvers:
+  - "@abhinavdahiya"
+  - "@smarterclayton"
+  - "@enxebre"
+  - "@deads2k"
+creation-date: 2019-11-06
+last-updated: 2019-11-06
+status: implemented
+---
+
+# Adding Baremetal IPI capabilities to OpenShift
+
+This enhancement serves to provide context for a whole slew of features
+and enhancements that will follow to make Baremetal IPI deployments via
+OpenShift a reality.
+
+At the time of this writing, code for some of these enhancements have already
+merged, some are in progress and others are yet to me implemented. References
+to all these features in different stages of development will be provided
+below.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Baremetal IPI deployments enable OpenShift to enroll baremetal servers to become
+Nodes that can run K8s workloads.
+The Baremetal Operator [1] along with other provisioning services (Ironic and
+dependencies) run in their own pod called "metal3". This pod is deployed by the
+Machine API Operator when the Platform type is `BareMetal`. The OpenShift
+Installer is responsble for providing all the necessary configs required for
+a successful deployment.
+
+## Motivation
+
+The motivation for this enhancement request is to provide a background for all the
+the subsequent enhancement requests for Baremetal IPI deployments.
+
+### Goals
+
+The goal of this enhancement request is to provide context for all the changes
```

Comment:
I think this doc should answer
1. comparison of metal3 to the AWS cloud provider.
2. components outside the cluster that have to be provisioned outside of a kube control plane, that will be used from inside the kube control plane.  (High level with links is fine).
3. new components inside the cluster that will speak to the outside metal3 components.  (High level with links is fine)
4. how internal apiserver LB, external apiserver LB, and ingresscontroller LBs will be managed an provisioned.  This needs to be pretty explicit
5. the network section from https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md#network-requirements with clear labeling about what, if anything, is managed on the cluster
6. top level item for the bootstrap host that the installer will use.

I may edit this comment to add more items as I have questions.